### DISCUSSION TITLE
Windows container doesn't seem to use pagefile.sys

### DISCUSSION THREAD

**Discussion by andivlad-talentia** [moby/moby#45808] [General]
I have a working scenario where i need to limmit the maximum memory usage in a windows container.  (image based on mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2022). Running in process isolation.

The expected behavior after the memory has been reached is to go into pagefile based on what i've been reading. This doesn't seem to be the case.

Following the steps described here https://thomasvanlaere.com/posts/2022/02/exploring-windows-containers-page-files/ (excellently written) i should see the pagefile starting to get used, and it doesn't:
```
docker run --isolation=process -m 2g --rm -it  mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2022 powershell.exe
```

Running `Get-WmiObject Win32_PageFileusage | Select-Object -Property *` yields
```
PSComputerName    : 6229068CC010
Status            : 
Name              : C:\pagefile.sys
CurrentUsage      : 118
__GENUS           : 2
__CLASS           : Win32_PageFileUsage
__SUPERCLASS      : CIM_LogicalElement
__DYNASTY         : CIM_ManagedSystemElement
__RELPATH         : Win32_PageFileUsage.Name="C:\\pagefile.sys"
__PROPERTY_COUNT  : 9
__DERIVATION      : {CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER          : 6229068CC010
__NAMESPACE       : root\cimv2
__PATH            : \\6229068CC010\root\cimv2:Win32_PageFileUsage.Name="C:\\pagefile.sys"
Caption           : C:\pagefile.sys
Description       : C:\pagefile.sys
InstallDate       :
PeakUsage         : 517
TempPageFile      : False
Scope             : System.Management.ManagementScope
Path              : \\6229068CC010\root\cimv2:Win32_PageFileUsage.Name="C:\\pagefile.sys"
Options           : System.Management.ObjectGetOptions
ClassPath         : \\6229068CC010\root\cimv2:Win32_PageFileUsage
Properties        : {AllocatedBaseSize, Caption, CurrentUsage, Description...}
SystemProperties  : {__GENUS, __CLASS, __SUPERCLASS, __DYNASTY...}
Qualifiers        : {dynamic, Locale, provider, UUID}
Site              :
Container         :
```

and then
```
Invoke-WebRequest "https://download.sysinternals.com/files/Testlimit.zip" -OutFile "TestLimit.zip"
Expand-Archive -Path ".\TestLimit.zip"
Set-Location -Path "TestLimit"
.\Testlimit64.exe -d 1 -c 2500 -accepteula
```

Container can only allocate the 2g memory and then the script get's stuck in retrying.

If i try to execute something else against the container i get:
`docker exec -it 6229068cc010 powershell
container 6229068cc010faf1724ae64267568026ac3796e43f44196c89b8bf57864adfb2 encountered an error during hcs::System::CreateProcess: failure in a Windows system call: The paging file is too small for this operation to complete. (0x5af)`

**Comment by corhere** [comment]
You may have better luck asking in https://github.com/microsoft/Windows-Containers as much of Windows' Host Compute Service is a black box to us Moby maintainers.

If I were to guess, I suspect that this is working as intended aside from `ERROR_COMMITMENT_LIMIT` having a misleading textual description: you requested that the container be limited to allocating 2GB of memory, and it has been limited to allocating 2GB of memory. In the article you linked, the author always sets the container memory limit to a larger value than the amount of memory allocated inside the container.
